invoiceregistry.go

```diff
@@ -81,18 +205,18 @@ func (i *invoiceRegistry) AddDebugInvoice(amt btcutil.Amount, preimage chainhash
 // AddInvoice adds a regular invoice for the specified amount, identified by
 // the passed preimage. Additionally, any memo or receipt data provided will
 // also be stored on-disk. Once this invoice is added, subsystems within the
-// daemon add/forward HTLCs are able to obtain the proper preimage required
-// for redemption in the case that we're the final destination.
+// daemon add/forward HTLCs are able to obtain the proper preimage required for
+// redemption in the case that we're the final destination.
 func (i *invoiceRegistry) AddInvoice(invoice *channeldb.Invoice) error {
 	ltndLog.Debugf("Adding invoice %v", newLogClosure(func() string {
 		return spew.Sdump(invoice)
 	}))
 
-	// TODO(roasbeef): also check in memory for quick lookups/settles?
-	return i.cdb.AddInvoice(invoice)
+	// We'll launch a new goroutine to notify all of our active listeners
+	// that a new invoice was just added.
+	go i.notifyClients(invoice, false)
```

Comment:
notify after write w/ no error?

Reply:
the invoices `AddIndex` wouldn't be set until after the db call

Reply:
Nice catch, fixed! Will also add a set of integration tests before the next round of review.